apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeArt.tsx

```diff
@@ -194,6 +202,144 @@ export default function ExcludeArt({
+      0
+    )
+    const numExcludedArt = artifacts.length - numIncludedArt
+    return { numExcludedArt, numIncludedArt, numUnlock, numLock }
+  }, [artifactIds, artExclusion, database])
+
+  const excludeArtifacts = () =>
+    window.confirm(
+      `Are you sure you want to exclude ${numIncludedArt} artifacts?`
+    ) &&
+    buildSettingDispatch({
+      artExclusion: [...artExclusion, ...artifactIds],
+      useExcludedArts: false,
+    })
+
+  const includeArtifacts = () =>
+    window.confirm(
+      `Are you sure you want to include ${numExcludedArt} artifacts?`
+    ) &&
+    buildSettingDispatch({
+      artExclusion: artExclusion.filter((i) => !artifactIds.includes(i)),
+      useExcludedArts: false,
+    })
+
+  const lockArtifacts = () =>
+    window.confirm(`Are you sure you want to lock ${numUnlock} artifacts?`) &&
+    artifactIds.map((id) => database.arts.set(id, { lock: true }))
+
+  const unlockArtifacts = () =>
+    window.confirm(`Are you sure you want to unlock ${numLock} artifacts?`) &&
+    artifactIds.map((id) => database.arts.set(id, { lock: false }))
+
+  return (
+    <Grid container spacing={1} alignItems="center">
+      <Grid item xs={12} sm={6} md={3}>
+        <Button
+          fullWidth
+          color="error"
+          disabled={!numLock}
+          onClick={unlockArtifacts}
+          startIcon={<LockOpenIcon />}
+        >
+          <Trans t={t} i18nKey="button.unlockrtifacts">
+            Unlock Artifacts
+          </Trans>
+          <SqBadge sx={{ ml: 1 }} color={numLock ? 'success' : 'secondary'}>
+            {numLock}
+          </SqBadge>
+        </Button>
+      </Grid>
+      <Grid item xs={12} sm={6} md={3}>
+        <Button
+          fullWidth
+          color="error"
+          disabled={!numUnlock}
+          onClick={lockArtifacts}
+          startIcon={<LockIcon />}
+        >
+          <Trans t={t} i18nKey="button.lockArtifacts">
+            Lock Artifacts
+          </Trans>
+          <SqBadge sx={{ ml: 1 }} color={numUnlock ? 'success' : 'secondary'}>
+            {numUnlock}
+          </SqBadge>
+        </Button>
+      </Grid>
+      <Grid item xs={12} sm={6} md={3}>
+        <Button
+          fullWidth
+          color="error"
+          disabled={!numIncludedArt}
+          onClick={excludeArtifacts}
+          startIcon={<BlockIcon />}
+        >
+          <Trans t={t} i18nKey="button.excludeArtifacts">
+            Exclude Artifacts
+          </Trans>
+          <SqBadge
+            sx={{ ml: 1 }}
+            color={numIncludedArt ? 'success' : 'secondary'}
+          >
+            {numIncludedArt}
+          </SqBadge>
+        </Button>
+      </Grid>
+      <Grid item xs={12} sm={6} md={3}>
+        <Button
+          fullWidth
+          color="error"
+          disabled={!numExcludedArt}
+          onClick={includeArtifacts}
+          startIcon={<ShowChartIcon />}
+        >
+          <Trans t={t} i18nKey="button.includeArtifacts">
```

Comment:
Now that we can include artifacts in this modal, the initial button name doesn't make a lot of sense.
<img width="841" alt="image" src="https://github.com/frzyc/genshin-optimizer/assets/36019388/4a0c7225-7044-4c70-aba8-6209600fb19f">
@PolomoPT any ideas?

Reply:
Oh related to translations, there's currently no text entry for including artifacts.